deployment/desktop/hosting-metadata.md

```diff
@@ -0,0 +1,198 @@
+---
+title: Hosting metadata for software applications 
+subsection: desktop
+description: >
+  How to host your GUI desktop application's metadata for yum/DNF or Flatpak in self- or Github-hosted repositories
+---
+
+# Hosting metadata for yum/DNF or Flatpak
+
+Metadata for distribution tools and GNOME Software must be refreshed
+periodically, such as when new versions of [packaged
+applications](desktop-software.html) are released for consumption by
+users.  In many cases, Linux distributions take care of the metadata
+extraction, generation, and hosting for these purposes.  However, this
+section provides information on how to host your own metadata in case
+you need to do so.
+
+## Install helper utilities
+
+Install the following software to help parse and generate the metadata needed for hosting:
+
+```console
+$ sudo dnf install createrepo_c libappstream-glib-builder flatpak-builder
+```
+
+## Generating yum repository metadata
+
+1. Run these commands to create the initial metadata for source
+   packages and the standard *x86_64* architecture:
+
+	```console
+$ createrepo_c --no-database --simple-md-filenames SRPMS/
+$ createrepo_c --no-database --simple-md-filenames x86_64/
+	```
+
+1. Next, generate the AppStream XML using the *appstream-builder*
+   tool:
+
+   ```console
+$ appstream-builder				\
+	--origin=yourcompanyname		\
+	--basename=appstream			\
+	--cache-dir=/tmp/asb-cache		\
+	--enable-hidpi				\
+	--max-threads=1				\
+	--min-icon-size=32			\
+	--output-dir=/tmp/asb-md		\
+	--packages-dir=x86_64/			\
+	--temp-dir=/tmp/asb-icons
+Scanning packages...
+Processing packages...
+Merging applications...
+Writing /tmp/asb-md/appstream.xml.gz...
+Writing /tmp/asb-md/appstream-icons.tar.gz...
+Writing /tmp/asb-md/appstream-screenshots.tar...
+Done!
+   ```
+
+	The actual build output will depend on your compose server
+    configuration. At this point you can also verify the application
+    is visible in the *yourcompanyname.xml.gz* file.
+
+1. Next, take the generated XML and the tarball of icons and add it to
+   the *repomd.xml* master document, so that GNOME Software
+   automatically downloads the content for searching:
+
+	```console
+$ modifyrepo_c					\
+	--no-compress				\
+	--simple-md-filenames			\
+	/tmp/asb-md/appstream.xml.gz		\
+	x86_64/repodata/
+$ modifyrepo_c					\
+	--no-compress				\
+	--simple-md-filenames			\
+	/tmp/asb-md/appstream-icons.tar.gz	\
+	x86_64/repodata/
+	```
+
+	Deploying this metadata allows GNOME Software to add the
+	application metadata the next time the repository is refreshed.
+
+## Hosting a DNF repository on Github
```

Comment:
Is this actually a proper workflow? Do many people host their repository on Github? I'm also not sure this is the correct place to insert guide for `git`. Some git section, or external git guide would be better. This is not exactly relevant to the topic.
Note that this shouldn't in any way by a `complete` guide, but merely a starting point and links to additional content. We expect the developers do learn themselves.

Reply:
This is also probably not a good idea to tell people to do, since it might not necessarily be something GitHub would appreciate...